tests/unit/s2n_alerts_protocol_test.c

```diff
@@ -573,6 +573,8 @@ int main(int argc, char **argv)
                         S2N_ERR_CLOSED);
                 EXPECT_FALSE(s2n_connection_check_io_status(receiver, S2N_IO_READABLE));
             }
+            /* Close the other end of pipe */
+            EXPECT_SUCCESS(s2n_io_pair_close_one_end(&io_pair, 1 - mode));
```

Comment:
I think the fix proposed in the closed [PR](https://github.com/aws/s2n-tls/pull/4831#discussion_r1793764059) will make it even more clear. I will do
```
EXPECT_SUCCESS(s2n_io_pair_close_one_end(&io_pair, S2N_PEER_MODE(mode)));
```

The variable `receiver` is defined as a `s2n_connection` which can't be passed in to this function.

Reply:
This `s2n_io_pair_close_one_end` is no longer needed, since I have implemented a logic to let `s2n_io_pair_close` to close io_pair's fds regardless whether one of them is closed or not.